mitiq/executor/executor.py

```diff
@@ -40,6 +40,8 @@
 FloatLike = [
     None,  # Untyped executors are assumed to return floats.
     float,
+    np.float32,
+    np.float64,
```

Comment:
I found the need to add these when running `make test`. Here is the result without those in FloatLike:
![image](https://github.com/user-attachments/assets/4b3287b2-6ac3-4f1f-9687-360f4e1b8fae)
And with those 2 lines, I don't see the errors:
![image](https://github.com/user-attachments/assets/7c4a7a3a-679b-4c04-8ffa-d7b1d66d661b)

@natestemen do you see this behavior too?

Reply:
I remembered why these two are in `FloatLike`. Since we are trying to assume things if a return type is not specified, we check the type of what is returned from `self.run()`. If we do not include them, then later checks (especially if we no longer look at `self._executor_return_type` as mentioned https://github.com/unitaryfund/mitiq/pull/2514#discussion_r1819858219

Reply:
I'm a little worried about this being needed now, despite me inspiring the change that required this to be added. I didn't foresee the consequences of moving from `_executor_return_type` to `manual_return_type`.

Imagine a situation where a user writes an executor that uses some machine learning stuff in `torch` and returns a [`torch.Tensor.float`](https://pytorch.org/docs/stable/generated/torch.Tensor.float.html). Since this type is not in `FloatLike` we will raise a `ValueError`! As far as I understand, this would be a workflow that would be broken by this PR.

As far as addressing it, I think we should rely on `_executor_return_type` for parsing the results. WDYT?

Reply:
I agree and think we are arriving back at executors requiring typehinting and that we cannot assume things are correct (though untyped executors will be treated as FloatLike.)

Reply:
Removed the user of `manual_return_type` and returned to `_executor_return_type` in https://github.com/unitaryfund/mitiq/pull/2514/commits/e3d8e2a7c3a9a8ab4516194e8429d54071656738